app/services/whatsapp/embedded_signup_service.rb

```diff
@@ -0,0 +1,49 @@
+class Whatsapp::EmbeddedSignupService
+  def initialize(account:, code:, business_id:, waba_id:, phone_number_id:)
+    @account = account
+    @code = code
+    @business_id = business_id
+    @waba_id = waba_id
+    @phone_number_id = phone_number_id
+  end
+
+  def perform
+    validate_parameters!
+
+    GlobalConfig.clear_cache
+
+    # Exchange code for user access token
+    access_token = Whatsapp::TokenExchangeService.new(@code).perform
+
+    # Fetch phone information
+    phone_info = Whatsapp::PhoneInfoService.new(@waba_id, @phone_number_id, access_token).perform
+
+    # Validate token has access to the WABA
+    Whatsapp::TokenValidationService.new(access_token, @waba_id).perform
+
+    # Create channel
+    waba_info = { waba_id: @waba_id, business_name: phone_info[:business_name] }
+    channel = Whatsapp::ChannelCreationService.new(@account, waba_info, phone_info, access_token).perform
+
+    # Setup webhook
+    Whatsapp::WebhookSetupService.new(channel, @waba_id, access_token).perform
```

Comment:
we can do this in an after create callback to ensure consistency with other channels like instagram and telegram...